src/include/migraphx/common.hpp

```diff
@@ -41,6 +41,9 @@ std::vector<shape::dynamic_dimension> compute_broadcasted_dyn_dims(shape s0, sha
 
 shape common_shape(const std::vector<shape>& shapes);
 
+std::vector<std::size_t> compute_common_lens(const std::vector<shape>& shapes);
+std::vector<shape::dynamic_dimension> compute_common_dyn_dims(const std::vector<shape>& shapes);
```

Comment:
Add docstring using `\brief @param @return` . It is pretty obvious what is doing but better to get into habit of adding docstring.

Reply:
I think VSCode will do that for you if you start typing `/**`.  I'm not sure what extension does that for me.